client/src/sagas/auth.js

```diff
@@ -1,20 +1,58 @@
-import { delay, put, takeLatest } from 'redux-saga/effects';
-import { LOGIN, LOGIN_ASYNC, REGISTER, REGISTER_ASYNC } from './types';
+import { push } from 'connected-react-router';
+import { put, takeLatest } from 'redux-saga/effects';
+import axios from 'axios';
 
-export function* loginAsync({ payload }) {
-  yield delay(1000);
-  yield put({ type: LOGIN, payload });
-}
+import { auth } from './types';
+import { alertActions } from './actions';
 
-export function* registerAsync({ payload }) {
-  yield delay(1000);
-  yield put({ type: REGISTER, payload });
+// logout
+export function* watchLogout() {
+  yield takeLatest(auth.LOGOUT, logout);
+}
+export function* logout() {
+  yield put({ type: auth.reducer.LOGOUT });
 }
 
+// login
 export function* watchLoginAsync() {
-  yield takeLatest(LOGIN_ASYNC, loginAsync);
+  yield takeLatest(auth.LOGIN, loginAsync);
+}
+export function* loginAsync({ payload }) {
+  try {
+    const res = yield axios.get(`/player/${payload}`);
+    yield put({
+      type: auth.reducer.LOGIN,
+      payload: { id: res.data.id, name: res.data.name },
+    });
+    yield put(alertActions.addAlert('Login successful!', 'success'));
+    yield put(push('/home'));
+  } catch (err) {
+    yield put({
+      type: auth.reducer.LOGIN_FAILED,
+      payload: err.response.data,
+    });
+    yield put(alertActions.addAlert(err.response.data, 'error'));
+  }
 }
 
+// register
 export function* watchRegisterAsync() {
-  yield takeLatest(REGISTER_ASYNC, registerAsync);
+  yield takeLatest(auth.REGISTER, registerAsync);
+}
+export function* registerAsync({ payload: { id, name } }) {
+  try {
+    const res = yield axios.post('/create/player', { id, name });
```

Comment:
how does this match the server endpoint?

https://github.com/joshprzybyszewski/cribbage/blob/428300bedd1dab3857e6234a4a9d28e713a287dd/server/server.go#L101-L102

Reply:
I set up a proxy in `package.json` to handle the base URL, is that what you mean?

Reply:
hmm ... no ... I think I don't understand how the vars called `id` and `name` populate the json fields `id` and `name` of the "create player request". How does that happen?

Reply:
The second arg of `axios.post` is a Javascript object which directly translates to JSON. In JavaScript, if you omit field names (as in `{ id, name }`), the fields automatically get named after the var. So that expressions yields some JSON:
```json
{
    id: “theID”,
    name: “theName”,
}
```